File: pkg/codegen/testing/test/testdata/kubernetes-template-pp/dotnet/kubernetes-template.cs
using System.Collections.Generic;
using Pulumi;
using Kubernetes = Pulumi.Kubernetes;

return await Deployment.RunAsync(() => 
{
    var argocd_serverDeployment = new Kubernetes.Apps.V1.Deployment("argocd_serverDeployment", new()
    {
        ApiVersion = "apps/v1",
        Kind = "Deployment",
        Metadata = 
        {
            { "name", "argocd-server" },
        },
        Spec = 
        {
            { "selector", new Kubernetes.Types.Inputs.Meta.V1.LabelSelectorArgs
            {
                MatchLabels = 
                {
                    { "app", "server" },
                },
            } },
            { "replicas", 1 },
            { "template", new Kubernetes.Types.Inputs.Core.V1.PodTemplateSpecArgs
            {
                Metadata = 
                {
                    { "labels", 
                    {
                        { "app", "server" },
                    } },
                },
                Spec = 
                {
                    { "containers", new[]
                    {
                        new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
                        {
                            Name = "nginx",
                            Image = "nginx",
                            ReadinessProbe = 
                            {
                                { "httpGet", 
                                {
                                    { "port", 8080 },
                                } },
                            },
                        },
                    } },
                },
            } },
        },
    });

});

File: pkg/codegen/testing/utils/testdata/kubernetes-template-pp/python/kubernetes-template.py
import pulumi
import pulumi_kubernetes as kubernetes

argocd_server_deployment = kubernetes.apps.v1.Deployment("argocd_serverDeployment",
    api_version="apps/v1",
    kind="Deployment",
    metadata={
        "name": "argocd-server",
    },
    spec={
        "selector": {
            "match_labels": {
                "app": "server",
            },
        },
        "replicas": 1,
        "template": {
            "metadata": {
                "labels": {
                    "app": "server",
                },
            },
            "spec": {
                "containers": [{
                    "name": "nginx",
                    "image": "nginx",
                    "readiness_probe": {
                        "http_get": {
                            "port": 8080,
                        },
                    },
                }],
            },
        },
    })
